source/fundamentals/gridfs.txt

```diff
@@ -0,0 +1,723 @@
+         :end-before: // end-open-download-stream-with-options-async
+
+   .. tab:: Synchronous
+      :tabid: open-download-stream-with-options-sync
+
+      .. literalinclude:: /includes/code-examples/GridFS.cs
+         :language: csharp
+         :dedent:
+         :start-after: // start-open-download-stream-with-options
+         :end-before: // end-open-download-stream-with-options
+   
+.. _csharp-gridfs-download_to_stream:
+
+Download to an Existing Stream
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use the ``DownloadToStreamAsync()`` or ``DownloadToStream()`` method to download the
+contents of a GridFS file to an existing stream. These methods accept the following parameters:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Parameter
+     - Description
+   * - ``id``
+     - The ``_id`` value of the file to download. 
+        
+       **Data type**: `BsonValue <{+new-api-root+}/MongoDB.Bson/MongoDB.Bson.BsonValue.html>`__
+   * - ``destination``
+     - The stream that the {+driver-short+} downloads the GridFS file to. This property's
+       value must be an object that implements the ``Stream`` class.
+        
+       **Data type**: `Stream <https://learn.microsoft.com/en-us/dotnet/api/system.io.stream?view=net-8.0>`__
+   * - ``options``
+     - *Optional.* An instance of the ``GridFSDownloadOptions`` class that specifies the
+       configuration for the download stream. The default value is ``null``.
+        
+       **Data type**: `GridFSDownloadOptions <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.GridFS.GridFSDownloadOptions.html>`__
+   * - ``cancellationToken``
+     - *Optional.* A token that you can use to cancel the operation.
+        
+       **Data type**: `CancellationToken <https://learn.microsoft.com/dotnet/api/system.threading.cancellationtoken>`__
+
+The following code example demonstrates how to download to an existing stream by
+performing the following actions:
+
+- Opens a file located at ``/path/to/output_file`` as a stream in binary write
+  mode
+- Retrieves the ``_id`` value of the GridFS file named ``"new_file"``
+- Calls the ``DownloadToStream()`` method and passes the ``_id`` value to download
+  the contents of ``"new_file"`` to a stream
+
+Select the :guilabel:`Asynchronous` or :guilabel:`Synchronous` tab to see the corresponding
+code.
+
+.. tabs::
+
+   .. tab:: Asynchronous
+      :tabid: download-to-stream-async
+
+      .. literalinclude:: /includes/code-examples/GridFS.cs
+         :language: csharp
+         :dedent:
+         :start-after: // start-download-to-stream-async
+         :end-before: // end-download-to-stream-async
+
+   .. tab:: Synchronous
+      :tabid: download-to-stream-sync
+
+      .. literalinclude:: /includes/code-examples/GridFS.cs
+         :language: csharp
+         :dedent:
+         :start-after: // start-download-to-stream
+         :end-before: // end-download-to-stream
+
+.. _csharp-gridfs-retrieve-file-info:
+
+Retrieve Files 
```

Comment:
"Retrieve" sounds like the wrong word here.

I think it should be "Find".

Retrieving is when you actually download the contents of the files.

Reply:
Are you going to change this?